### ISSUE TITLE
makenek.inc bug with PGI compilers

### ISSUE THREAD

**Issue by RonRahaman** [Nek5000/Nek5000#298]
There is a bug in makenek.inc that can often cause problems with the PGI compilers.  In particular, it fails to identify the PGI compiler correctly.  

This portion of makenek.inc is one attempt to detect the compiler:
``` bash
if [ $F77ok -eq 0 ]; then
  F77comp_=`$F77 -show 2>/dev/null 1>.tmp || true`
  F77comp=`cat .tmp | awk '{print $1}' | awk -F/ '{print $NF}' || true`
  if [ -f "`which $F77comp 2>/dev/null`" ]; then
    F77ok=1
  fi
fi
```
If you have set 'F77=pgfortran`, then the 'pgfortran -show` command will do the following (according to the manpage):
```
       -show     Produce help information describing the current pgfortran configuration.
```
And then `F77comp` will eventually be set to this garbage:
```
Known ACC901FLAGS ACCAUTOPAR ACCBUILDLIB ACCCACHE ACCCGDEF ACCCGFLAGS ACCDEBUG ACCDEF ACCDEFDEF ACCDEFINES ACCELFLAG ACCELFLAG1 ACCELS ACCEXTRACTLIB ACCEXTRACTLIBSET ACCEXTRACTLIBSW ACCEXTRACTTOC ACCFEFLAGS ACCFLAGS ACCLIB ACCLIB2 ACCLIBSUFF ACCLINK ACCLLVM ACCMPSUFFIX ACCMULTI ACCMULTIONLY ACCPREFIX ACCPRESTDINC ACCRELOC ACCREQUIRED ACCRPATH ACCTESLAONLY ACCWAIT ACTTPVAL ADDEHFRAME ALIASMODE ALLOCATABLE03 ANYCUF AOPT APPENDMOD APPENDTOOL AS ASARGS ASDIR ASMCOMMAND ASMONLY ASMSUFFIX ASMSUFFIXES ASTYPE ATHLON AUTOINLINE AUTOLEVELS AUTOMAXFUNCSIZE AUTOMAXTOTALSIZE AVX AVX2 BADCUDALIB BLANKCUDALIB BLTINLIBS BMODE BOUNDSCHECK C99 CARGS CC60 CCOMP CCOMPDIR CEXT CEXTARGS CGARGS CGMANPIN CHEADERSUFFIX CHECKCUDALIB CLTPVAL COMPBASE COMPBASE32 COMPBASE64 COMPBBIN COMPBIN COMPGCCINCDIR COMPGLIBDIR COMPGLIBLFDIR COMPILEONLY COMPINCDIR COMPLIB COMPLIBDIR COMPLIBLF COMPLIBLFDIR COMPLIBMP COMPLIBOBJ COMPLIBSO COMPLIBSTATIC COMPLIBSUBDIR COMPSUBBIN COMPSYS COMPUTECAP COMPUTECAPS COMPVER CONCUR CONCURCOMPILE CONCURSW CPLUS32LDARGS CPLUS64LDARGS CPP CPP1ARGS CPP1MANPIN CPP2ARGS CPPASC CPPASCPP CPPCOMPCOMMANDS CPPEXTCOMMANDS CPPHEADERSUFFIX CPPLDARGS CPPLIB CPPSTDLIBS CPPSUFFIX CRTL CSUFFIX CUCOMMANDS CUDA_ATTACH CUDAEMU CUDAINC CUDALIBDIR CUDALIBDIRNAME CUDALIBINSTALL CUDALIBSTATIC CUDANEEDED CUDAROOT CUDARPATH CUDARPATHNEEDED CUDARTLIBS CUDARTNEEDED CUDARTSTATIC CUDASITELIB CUDAVERSION CUDAVERSIONDEFINE CUDAVSN CUFSUFFIX CUFX CULIBOSSTATIC CZCCFF DDEFACCEL DDEFNOOPENMP DDEFNOSGIMP DEBUG DEBUG2 DEBUG2DEF DEBUGFLAG DEBUGFLAG_64 DEBUGSET DEF901ACC DEFACC DEFACCEL DEFACCPRESTDINC DEFACCRELOC DEFAS DEFASARGS DEFASDIR DEFASTYPE DEFAULTCAPFLAG DEFAULTCHEADERS DEFAULTCPPASCPP DEFAULTCPPHEADERS DEFAULTLLVM DEFBMODE DEFC99 DEFCACHESIZE DEFCARGS -x -x -x -tp DEFCCOMPDIR DEFCEXTARGS -x -x -x -tp DEFCGMANPIN DEFCOMMON -x -x -x -tp DEFCOMPUTECAP DEFCOMPUTECAPS DEFCPP1MANPIN DEFCPPARGS -x -x -x -tp DEFCPPDEFS DEFCPPINC DEFCPPLDARGS DEFCPPLIB DEFCPPLIBS DEFCPPSTDLIBS DEFCUDARPATH DEFCUDAVERSION DEFDEFACCRELOC DEFDEFDEF4ACCRELOC DEFDEFDEFACCRELOC DEFDEFS DEFDEFUNSIGNED DEFF901ARGS -quad -x -x DEFF901MANPIN DEFF902ARGS -x -x -x -x -tp DEFF902COMMON -x -x -x -x -tp DEFF90COMMON -quad -x -x DEFF90EXTARGS DEFFARGS -x -x -x -tp DEFFEXTARGS -x -x -x -tp DEFFLUSHVAL DEFFTNRTL DEFGNU DEFGNU2 DEFGNUC DEFHOST DEFHPF1ARGS -quad -x -x DEFHPF2ARGS -x -x -x -x -tp DEFHPFEXTARGS DEFINSTLIB DEFKEEPOBJ DEFKEEPPREPRO DEFLCUDAFORLIB DEFLCUDALIB DEFLD DEFLDARGS DEFLDDIR DEFLDENDUP DEFLDSTARTUP DEFLIBDIR DEFLINKFILES DEFLINKOUT DEFMANMCUDALIB DEFMANOBJ DEFMULTIONLY DEFMUNCHLDSTARTUP DEFMVDIR DEFNOOPENMP DEFNOSGIMP DEFPRED DEFPREPRO DEFPROFLIB DEFSMARTARGS DEFSTDDIRS DEFSTDINC DEFSTDINC32 DEFSTDINC64 DEFSTDLIBS DEFSTDOBJDIR DEFSTDPATHS DEFTESLAONLY DEFTPVAL DEFUSESSELIBS DLLIB DOACCLINK DOCCFF DODWARF DOPRELINK DRIVERCOMMAND DRIVERDIR DRIVERHOST DRIVERLANG DRIVERNAME DRYRUN DRYSTDINC dummy DYNAMICLINK ECHONAME EHFRAME ENDNUMADYNAMIC EXE EXTENSION EXTRACOMPLIBDIR EXTRACT EXTRACTINLINE EXTRACTLIB EXTRACTLIBDEF EXTRACTLIBSET EXTRACTLIBVAL EXTRACTONLY EXTRACTTOC EXTRACUDAFORLIB EXTRAF901ARGS EXTRAF902ARGS EXTRAHPF1ARGS EXTRAHPF2ARGS F77SUFFIX F901 F901ARGS F901DEFARGS F901MANPIN F902 F902ARGS F90E F90EXT F90SUFFIX F90X FARGS FBADCUDALIB FBLANKCUDALIB FCGARGS FCOMP FCOMPCOMMANDS FEXT FEXTCOMMANDS FFEARGS file files FIPACGARGS FIPAVAL FIPAVALUE FLUSHVAL FMA3 FMA4 FNEEDCUDA FORCEDBGLLVM FORCELILLVM FORCELLVM FORTRANMPI FPIC FTNALT FTNMAIN FTNRTL FTNUNIXALT G77DIR G77LIBDIR G77LIBLIB GBASE GCCDIR GCCINC GCCINC32 GCCINC64 GCCVERSION GENCCFF GPPDIR GPPDIR32 GPPDIR64 GSYS GSYSFLAG HANDLEOTHER HASMSMPIFLIBNAME HASMSMPIFLIBNAME32 HASMSMPIFLIBNAME64 HCOMPDIR helpvalues HOSTCPP1ARGS HOSTOBJSUFFIX HPF1 HPF1ARGS HPF2 HPFE HPFEXT HPFSUFFIX HPFX HUGELIB HUGEN I8STORAGE IEEE IGNORECUDALIB ILPALIGN IMPSW INFORM INFOTPVAL INITCUDAOBJ INLINE INLINEARGLIB INLINEARGS INLINECARGS INLINELIST INLINESIZE input INTRINIL INTRINLEVEL IPA14 IPA14r IPA66 IPA89 IPAADD IPAADDINLINE IPAADDSW IPAC2ARGS IPACARGS IPACGARGS IPADIR IPAEXPORT IPAEXPORTVAL IPAFASTVESTHELP IPAFASTVESTSW IPAFASTVESTX89 IPAIGNORE0 IPAIGNORE1 IPAIGNORE2 IPAINLINE IPAINLINELIST IPALD IPALIBCSW IPALINK IPALINKENABLE IPANOARG IPARECOMPILE IPARGS IPAVAL IPAVALUE IPAVERBOSITY IPNOP IPOFILE ISFTN KEEPASM KEEPCCFF KEEPIL KEEPOBJ KEEPPREASM KEEPPREPRO KEEPTEMP LC LCRT1 LCRTB LCRTBDEF LCRTE LCRTI LCRTN LCUDAFORLIB LCUDALIB LD LDARGS LDDIR LDINPUTPOST LDINPUTPRE LDLIBARGS LDLIBS LDSO LDSWITCHES LEGACY LEVENTI LFC LIBDIR LIBMEMIL LIBSSW LIBSUFFIX LIBSW LIBSW_NOLIB LINITACC LINITMP LINKNOBJSUFFIX LINKOBJSUFFIX LINKSUFFIX LINUXCOMMON LLVMFLAGS LM LMP LMP_ONLYMP LMPILIB LMPIPROFLIB LNSPGC LOCALBINDLIB LOCALBINDPATHS LOCALCOMPLIB LOCALCOMPLIBLF LOCALDEFS LOCALRC LPROFLIB LPROFO LRTLIB LRTLIBNAME LSCALAPACK LTHREAD LTRACE MAJOR MALIGN MANAGED MANMCUDALIB MANOBJ MCUDADEF MCUDAEMUDEF MCUDALIB MICLINKARGS MPDEFCPP MPDEFCPPDEF MPFAIR MPFAIRDEF MPI_EXTRAFTNOBJ MPIBASEDIR MPIBASELIB MPIINC MPILIB MPILIBDIR MPILIBDIRNAME MPILIBINC MPILIBNAME MPILIBVER MPIMSINC MPIMSLIB MPIMSLIBNAME MPISGIDIR MPISGIINC MPISGILIB MPISGIROOT MPITYPE MPIUDIR MPSET MPSETCPP MPSETDEF MPSETDEFCPP MSGTPVAL MSLINK MSMPI_INC MSMPI_LIB32 MSMPI_LIB64 MSMPIDIR MSMPIFLIBNAME MSMPIINC MSMPILIB32 MSMPILIB64 MVECT MYLOCALRC NEEDACCLIB NEEDCUDA55 NEEDCUDA60 NEEDCUDA65 NEEDCUDA70 NEEDCUDA80 NKEY NOASM NOATTACHOBJ NOBOUNDSCHECK NOFRAME NOILM NOLLALIGN NOOPENMP NOPGIRPATH NORPATH NOSGIMP NOSIMD256 NOSTARTUP NOSTDCPPLIBS NOSTDFTNLIBS NOSTDLIBDIRS NOSTDLIBPATHS NOSTDLIBS NOUNIFIED NOUNIFIEDMSG NOWARN NULLINPUT NUMADYNAMIC NUMALIBNAME NUMALIBS NUMAOBJS NUMAOBJSDEF NVILINKARGS OBJDIR OBJSUFFIX OBJSUFFIXES OEM_INFO OFEDLIBDIR OLDSTDINC OMPINIT OMPINITOBJ OMPLINKFILE OPTLEVEL OPTLEVELDEF OPTLEVELINITDEF OPTLEVELINITDEF1 OPTLEVELMAX OPTLEVELMIN OUTFILE PAPIDIR PATH PERSONALITY PGATM PGF77LIB PGF90LIB PGIINCDIRENV PGILD PGILIBSDIRENV PGILIBSENV PGINUMALIBS PGIOPTLINCDIR PGIOPTLLIBS PGIOPTLLIBSDIR PGISTATIC PGISTATICX PGIUNSTATIC PGIUNSTATICX PGLEN PGSYS PINOBJ POSTOPTIONS PREASM PREASMCMD PREASMSUFFIX PREFIX PREOPTIONS PREPRO PREPROARGS PREPROCONLY PREPRODIR PREPROSUFFIX PROFFLAG PROFILE PROFILESW RADEONBIN RADEONBINS realfiles SCALAPACK_ACML SCALARSSE SETDOCCFF SIMD256 SITEASMPPSUFFIX SITEASMSUFFIX SITECPPSUFFIX SITECSUFFIX SITECUFPPSUFFIX SITECUFSUFFIX SITEDEF SITEF77PPSUFFIX SITEF77SUFFIX SITEF90PPSUFFIX SITEF90SUFFIX SITEHPFPPSUFFIX SITEHPFSUFFIX SITEINC SITELIB SITELIBSUFFIX SITEOBJSUFFIX SITERPATH SITERPATHS SMART SMARTARGS SMARTDIR SRCFILES SRCSUFFIX SSE SSE2 SSE3 SSE41 SSE42 SSE4A SSEDEFS SSSE3 STATICLIB status STDDEF STDINC STDOBJDIR STDPRED STDRPATH STDRPATHLF STDRPATHMP STDRPATHSO STRICTBMODE switches SYS32 SYS64 SYSACCELS SYSACCLIBS SYSACCOBJS SYSV_PROF TARGET TARGETARCH TARGETCOMMON THROW TIMEOBJ TIMER TOOLKITFLAG TPVAL TRACEBACK TRACEBACKSW USECUDAROOT USERASMPPSUFFIX USERASMSUFFIX USERCPPSUFFIX USERCSUFFIX USERCUFPPSUFFIX USERCUFSUFFIX USERF77PPSUFFIX USERF77SUFFIX USERF90PPSUFFIX USERF90SUFFIX USERHPFPPSUFFIX USERHPFSUFFIX USERLIBSUFFIX USEROBJSUFFIX USESSELIBS USETPVAL USRDDEF USRDEFDEF USRFTNRTL USRINC USRRPATH USRSTDINC VECT VECTSSE VERBOSE VERSION WRSTR X86_64 X86_64TARGET XTRAS ZCDEF ZCFLAG ZCLIBS Known Extract C Fortran Fortran High Preprocess Preprocess Compile C Fortran Fortran hpfx Fortran Assembler Preprocess IPA IPA Link Linker
```

Somewhere in there is the token "file".   *This is actually a program (/usr/bin/file).*  Hence, the command `if [ -f "`which $F77comp 2>/dev/null`" ]` will succeed because `which $F77comp` will return `/usr/bin/file`, which is a file that exists.  

Right now, makenek does *not* succeed on my workstation, and I don't know why it has suceeded in the past.  Perhaps the output from `pgfortran -show` is different on my machine.  I.e., on most machines `pgfortran -show` does not include the token "file".

**Comment by stgeke** [comment]
Yes we rely on the assumption that only the MPI wrappers have a `-show` or `-showme` argument. At the moment I don't have a good idea how to fix this.

**Comment by RonRahaman** [comment]
Understood.  For now, I'm fine if I just compile with IFMPI=false, F77=mpif77, and CC=mpicc, where the MPI compilers are PGI installations.  The GPU work is running in serial on CPU now, but we're almost to the point where we can run with MPI on CPU, too.

**Comment by kent0** [comment]
@RonRahaman @stgeke I submitted a PR to fix the bug: #334
